views/src/js/posts_list.js

```diff
@@ -0,0 +1,107 @@
+function ListItemHeader(props) {
+    return <h3>{props.post_header}</h3>;
+}
+
+function ListItemBody(props) {
+    const post = Object.entries(props.post);
+
+    return (
+        <ul>
+            {
+                post.map(post_field => <li key={post_field[0]}><b>{post_field[0] + ": "}</b>{post_field[1]}</li>)
+            }
+        </ul>
+    );
+}
+
+function ListItem({post}) {
+
+    return (
+    <div>
+        <div id="list-item-header">
+            {
+                <ListItemHeader key={post.unique_id} post_header={post.post_category}/>
+            }
+        </div>
+        <div id="list-item-body">
+            {
+                <ListItemBody key={post.unique_id} post={post}/>
+            }
+        </div>
+    </div>
+    );
+}
+
+function ActionLink({value, page, setPage, operation}) {
+    function handleClick() {
+        configurations.page = operation(page);
+
+        if (configurations.page !== -1) {
+            fetchPosts(createFiltersLine(configurations))
+            .then(data => {
+                if (data.length === 0) {
+                    configurations.page -= 1;
+                }
+                return data;
+            }).then(data => {
+                if (configurations.page !== page) {
+                    renderList(data, postsContainer);
+                }
+                setPage(configurations.page);
```

Comment:
what if you cant fetch the data. need maybe error notification, cuz user don't know what happend at all